src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -208,5 +227,130 @@ static void VerifyMemberOnlyApplyToTypesOrStrings (SymbolAnalysisContext context
+				}
+			}
+
+			for (int i = 0; i < method.TypeParameters.Length; i++) {
+				var methodTypeParam = method.TypeParameters[i];
+				var overriddenTypeParam = overriddenMethod.TypeParameters[i];
+				var methodParamHasDam = methodTypeParam.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var methodTypeParamAttribute);
+				var overriddenParamHasDam = overriddenTypeParam.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var overriddenTypeParamAttribute);
+				if ((methodParamHasDam && overriddenParamHasDam && !DamAttributesMatch (methodTypeParamAttribute!, overriddenTypeParamAttribute!))
+					|| (methodParamHasDam ^ overriddenParamHasDam)) {
+					context.ReportDiagnostic (Diagnostic.Create (
+						DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersMismatchOnGenericParameterBetweenOverrides),
+						methodTypeParam.Locations[0],
+						methodTypeParam.GetDisplayName (), method.GetDisplayName (), overriddenTypeParam.GetDisplayName (), overriddenMethod.GetDisplayName ()));
+				}
+			}
+
+			var basestTypeSymbol = context.Symbol.ContainingType;
+			while (basestTypeSymbol?.BaseType != null) {
+				if (basestTypeSymbol.Name == "Type" && basestTypeSymbol.ContainingNamespace.Name == "System") {
+					var methodHasDam = method.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var methodAttribute);
+					var overriddenMethodHasDam = overriddenMethod.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var overriddenMethodAttribute);
+					if ((methodHasDam && overriddenMethodHasDam && !DamAttributesMatch (methodAttribute!, overriddenMethodAttribute!))
+						|| (methodHasDam ^ overriddenMethodHasDam)) {
+						context.ReportDiagnostic (Diagnostic.Create (
+							DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersMismatchOnImplicitThisBetweenOverrides),
+							method.Locations[0],
+							method.GetDisplayName (), overriddenMethod.GetDisplayName ()));
+					}
+				}
+				basestTypeSymbol = basestTypeSymbol.BaseType;
+			}
+		}
+
+		static void VerifyDamOnInterfaceAndImplementationMethodsMatch (INamedTypeSymbol type, SymbolAnalysisContext context)
+		{
+			foreach (var (interfaceMember, implementationMember) in type.GetMemberInterfaceImplementationPairs ()) {
+				if (implementationMember is IMethodSymbol implementationMethod
+					&& interfaceMember is IMethodSymbol interfaceMethod)
+					VerifyMethodDamAttributesMatch (implementationMethod, interfaceMethod, context);
+			}
+		}
+
+		static void VerifyDamOnPropertyAndAccessorMatch (IMethodSymbol methodSymbol, SymbolAnalysisContext context)
+		{
+			if (!(methodSymbol.AssociatedSymbol?.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var propertyAttribute) == true)
+				|| (propertyAttribute.ConstructorArguments.Length == 1 && propertyAttribute.ConstructorArguments[0].Value is (int)DynamicallyAccessedMemberTypes.None))
+				return;
```

Comment:
This will probably run for event accessors as well (`add`, `remove`, ...) - probably not a problem, but maybe to keep the analyzer more strict, we should check that the method in question is getter/setter first.